File: push/src/main/java/com/xiaomi/push/mpcd/CDActionProviderHolder.java
package com.xiaomi.push.mpcd;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/push/mpcd/CDActionProviderHolder.class */
public class CDActionProviderHolder {
    private static volatile CDActionProviderHolder sInstance;
    private CDActionProvider mCDActionProvider;

    public static CDActionProviderHolder getInstance() {
        if (sInstance == null) {
            synchronized (CDActionProviderHolder.class) {
                try {
                    if (sInstance == null) {
                        sInstance = new CDActionProviderHolder();
                    }
                } catch (Throwable th) {
                    throw th;
                }
            }
        }
        return sInstance;
    }

    public CDActionProvider getCDActionProvider() {
        return this.mCDActionProvider;
    }

    public void setCDActionProvider(CDActionProvider cDActionProvider) {
        this.mCDActionProvider = cDActionProvider;
    }
}

File: legacy-runtime/src/main/java/com/xiaomi/push/mpcd/CDActionProviderHolder.kt
package com.xiaomi.push.mpcd

class CDActionProviderHolder {
    private var mCDActionProvider: CDActionProvider? = null

    fun getCDActionProvider(): CDActionProvider? = mCDActionProvider

    fun setCDActionProvider(cDActionProvider: CDActionProvider?) {
        mCDActionProvider = cDActionProvider
    }

    companion object {
        @Volatile
        private var sInstance: CDActionProviderHolder? = null

        @JvmStatic
        fun getInstance(): CDActionProviderHolder {
            if (sInstance == null) {
                synchronized(CDActionProviderHolder::class.java) {
                    if (sInstance == null) {
                        sInstance = CDActionProviderHolder()
                    }
                }
            }
            return sInstance!!
        }
    }
}
